### ISSUE TITLE
ImGUI on viewmodel doesn't properly find cursor position

### ISSUE THREAD

**Issue by vaqxai** [wyozi-gmod/imgui#35]
**
![image](https://user-images.githubusercontent.com/19872682/134334074-4bca02e8-3a1a-4228-8ad9-b2c97cdea7b9.png)
**

the white crosss is imgui.CursorPos as reported by imgui, the mouse is the actual cursor pos
the imgui panel is attached to the player's viewmodel
at screen middle they're the same

i'm thinking this might be an issue with EyePos being too close to the viewmodel?

**Comment by wyozi** [comment]
Yeah, viewmodel tends to be a bit difficult to map the mouse position of. I'm not sure what would be the best approach here, but if you figure out something it would be an amazing addition to the library

**Comment by dhkatz** [comment]
I've been trying to figure this out lately. When the weapon's view model FOV matches the player's FOV then the cursors line up as expected. When they are different is when the positions diverge.

I've been trying to figure out the math to transform/scale different parts of the hit position in `imgui.Start3D2D` but no luck so far.

The only workaround I know now is to just force the player's FOV to be the weapon's FOV (or the other way around). But ideally we should figure out a way to automatically handle FOV differences (perhaps in a function like `imgui.Weapon3D2D`)

**Comment by dhkatz** [comment]
I've created what I believe to be a fix for this https://github.com/wyozi-gmod/imgui/pull/42
